### PR TITLE
sd-journal.c: port to extract_first_word

### PR DESCRIPTION
**Pull request by ssahani**
Testcase: test-journal-stream Success

### CODE REVIEW

File: src/journal/sd-journal.c [hunk 1]
```diff
@@ -1040,20 +1038,23 @@ _public_ int sd_journal_test_cursor(sd_journal *j, const char *cursor) {
...
+        for(;;) {
                 _cleanup_free_ char *item = NULL;
-                sd_id128_t id;
                 unsigned long long ll;
+                sd_id128_t id;
                 int k = 0;
 
-                if (l < 2 || word[1] != '=')
-                        return -EINVAL;
+                r = extract_first_word(&cursor, &item, ";", 0);
```
Comment (evverx) [inline src/journal/sd-journal.c]: maybe

```
r = extract_first_word(&cursor, &item, ";", EXTRACT_DONT_COALESCE_SEPARATORS);
```

?

Comment (poettering) [inline src/journal/sd-journal.c]: Yes, @everx is right.

File: src/journal/sd-journal.c [hunk 2]
```diff
@@ -1040,20 +1038,23 @@ _public_ int sd_journal_test_cursor(sd_journal *j, const char *cursor) {
...
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse cursor: %s", cursor);
```
Comment (poettering) [inline src/journal/sd-journal.c]: Please do not log from this function, it's a library call that is not supposed to log on its own, the caller should. It doesn't log in any other error case, and shouldn't in this one either. Just return r directly...

**Comment by ssahani** [comment]
Fixed thanks !

**Pull request merged**